src/test/java/teammates/test/cases/datatransfer/AccountAttributesTest.java

```diff
@@ -139,6 +139,46 @@ public void testLegacyAccountEntityToAttributes() {
 
     }
 
+    @Test
+    public void testGetCopy_typicalData_shouldCreateDeepCopy() {
+        AccountAttributes account = createValidAccountAttributesObject();
+
+        AccountAttributes copy = account.getCopy();
+
+        assertNotSame(account, copy);
+        assertNotSame(account.studentProfile, copy.studentProfile);
+        assertFalse(account.isInstructor);
+
+        assertEquals(account.googleId, copy.googleId);
+        assertEquals(account.name, copy.name);
+        assertEquals(account.institute, copy.institute);
+        assertEquals(account.email, copy.email);
+    }
+
+    @Test
+    public void testGetCopy_ifAllFieldsNull_shouldCreateDeepCopy() {
+        AccountAttributes account = AccountAttributes.builder()
+                .withGoogleId(null)
+                .withName(null)
+                .withEmail(null)
+                .withInstitute(null)
+                .withIsInstructor(false)
+                .withStudentProfileAttributes(null)
+                .build();
+
+        AccountAttributes copy = account.getCopy();
+
+        assertNotSame(account, copy);
+        assertFalse(account.isInstructor);
+
+        assertNull("student profile should be null", copy.studentProfile);
+
+        assertEquals(account.googleId, copy.googleId);
+        assertEquals(account.name, copy.name);
+        assertEquals(account.institute, copy.institute);
+        assertEquals(account.email, copy.email);
```

Comment:
Any chance of extracting the common parts as `verifyDeepCopy(original, copy)`? @xpdavid  what do you think?

Reply:
Actually, using `assertNull(xxx, xxx)` should be more appropriate here since we are testing `null` scenario. If we are doing this, there is no code duplication and it is a bit overkill to extract the common part.